src/sql/parts/connection/connectionDialog/connectionWidget.ts

```diff
@@ -176,6 +190,50 @@ export class ConnectionWidget {
 		return false;
 	}
 
+	private _updateDatabaseNames() : void {
+		if (this.serverName) {
+			// connect to server and fetch database names
+			if (this.authenticationType === Constants.integrated) {
+				this._databaseNameInputBox.setOptions(['Loading...'], 0);
+				this._callbacks.onUpdateDatabaseNames().then(result => {
+					if (result) {
+						this._databaseNameOptions = [this.DefaultDatabaseGroup];
+						for (let database of result) {
+							this._databaseNameOptions.push({
+								id: '',
+								name: database,
+								parentId: undefined,
+								color: undefined,
+								description: undefined,
+							});
+						}
+						this._databaseNameInputBox.setOptions(this._databaseNameOptions.map(d => d.name), 0);
+					} else {
+						this._databaseNameInputBox.setOptions([this.DefaultDatabaseGroup, this.LoadingDatabaseGroup].map(d => d.name), 0);
+					}
+					this._serverNameInputBox.enable();
```

Comment:
>this._serverNameInputBox [](start = 5, length = 24)

is there any concern that the dialog may be closed while this async task is running?  i.e., have you verified this task is properly canceled in this scenario?